src/Microsoft.ML.Data/Transforms/Normalizer.cs

```diff
@@ -319,7 +319,7 @@ public ColumnFunctionAccessor(ColumnInfo[] infos)
         }
 
         /// <summary>An accessor of the column functions within <see cref="_columns"/>.</summary>
-        internal readonly IReadOnlyList<IColumnFunction> ColumnFunctions;
+        public readonly IReadOnlyList<IColumnFunction> ColumnFunctions;
```

Comment:
Agree. And I don't like that we reference the ColumnFunctions by indexes. A dictionary would be better. 

---
In reply to: [232817552](https://github.com/dotnet/machinelearning/pull/1594#discussion_r232817552) [](ancestors = 232817552)

Reply:
I still don't like it. Again, we have `public (string input, string output)[] Columns`, and we ALSO have a dictionary with model parameters. Why not merge the two collections into an `ImmutableArray<ColumnInfo> Columns` ?

---
In reply to: [232912516](https://github.com/dotnet/machinelearning/pull/1594#discussion_r232912516) [](ancestors = 232912516,232817552)

Reply:
Removed Columns and exposed the non-instantiable ColumnInfos where all the fields are readonly. ColumnFunction is still there;  It can be removed, but I'd rather do that on a separate PR; if that's ok. 

---
In reply to: [233682013](https://github.com/dotnet/machinelearning/pull/1594#discussion_r233682013) [](ancestors = 233682013,232912516,232817552)